[MRG] MAINT: refactor viz functions

jasmainak opened the pull request:
as discussed in the roadmap.

TODO:

- [x] add tests and file `test_viz.py`
- [x] update `api.rst`

rythorpe commented on hnn_core/viz.py:
```diff
@@ -0,0 +1,165 @@
+    return ax.get_figure()
+
+
+def plot_hist_input(net, ax=None, show=True):
```
What do you think of creating a more general function that can plot any spike type (i.e., not just input spikes): `plot_spikes_hist(net, ax=None, spike_types=None, show=True)`? `spike_types` could be entered by the user as a dict of lists s.t. the elements of each list specify which spike types are included in a given category of the histogram. E.g., `spike_types={'proximal inputs': ['evprox_1', 'evprox_2'], 'distal inputs': ['evdist_1']}` would plot the histogram we expect to see for the default.json parameter set. This would establish a much more flexible API that explicitly defines which types (e.g., 'evprox', 'evdist', distal 'common' feed spikes, etc.) are being plotted as discussed in #135.

jasmainak replied:
I agree with what you say, do you want me to do this in this PR itself? Or should we leave it for a new contributor as discussed in #135 ? 

I can also do it in another PR myself if I'm getting bored on a Friday night ... I feel this current PR is more of "moving things around"

rythorpe replied:
@jasmainak I guess let's save my suggestions for another PR and/or contributor. It should be pretty straight-forward given this PR.

jasmainak replied:
Feel free to copy the comment or add a pointer to it in #135 :)

rythorpe commented on hnn_core/viz.py:
```diff
@@ -0,0 +1,165 @@
+    return ax.get_figure()
+
+
+def plot_spikes(spikes, ax=None, show=True):
```
To complement `plot_spikes_hist()`, this function could also be renamed `plot_spikes_raster()`. Since the vast majority of use-cases for viewing a raster plot are for output spikes, I think it makes sense to leave the hard-coded spike types (i.e., cell_types = `['L5_pyramidal', 'L5_basket', 'L2_pyramidal', 'L2_basket']`) as-is.

jasmainak replied:
sounds good, I'll change the name.

jasmainak commented:
@rythorpe I addressed your comment in https://github.com/jonescompneurolab/hnn-core/pull/133/commits/b72770bb49e4c3ec7ebe718b9027fac7f9405e8b. Feel free to rebase and merge or squash and merge if you're happy
